### PR TITLE
refactoring: factor out analysis in goto-analyzer

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 632f8269c6a2379c8725f0d3beb7c8ffef11e502).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89436581

File: src/goto-analyzer/static_verifier.cpp [hunk 1]
```diff
@@ -38,229 +38,224 @@ bool static_verifier(
   messaget m(message_handler);
   m.status() << "Checking assertions" << messaget::eom;
```
Comment (peterschrammel) [inline src/goto-analyzer/static_verifier.cpp:228]: Can we split this function please? It has already 250 lines.

Comment (kroening) [reply src/goto-analyzer/static_verifier.cpp:228]: Ok, split

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b5ef0cbb5fe42c27d73b7b7b13dd65e8769f1245).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89496781

**Review by martin-cs — APPROVED** [review]
Seems to simplify things.

**Pull request merged**